### PR TITLE
[Kotlin Migration] DeckDropDownAdapter

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
CI failure

**Comment by Akshay0701** [comment]
```
com.ichi2.anki.servicemodel.UpgradeVolumeButtonsToBindingsTest > test_preferences_not_opened_happy_path[1: isValid(TestData(affectedPreferenceKey=gestureVolumeDown, keyCode=25, unaffectedPreferenceKey=gestureVolumeUp, binding=com.ichi2.anki.reviewer.MappableBinding@19753a))={1}] FAILED
    java.lang.AssertionError: 
    Expected: iterable containing ["preferenceUpgradeVersion"]
         but: No item matched: "preferenceUpgradeVersion"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
        at com.ichi2.anki.servicemodel.UpgradeVolumeButtonsToBindingsTest.test_preferences_not_opened_happy_path(UpgradeVolumeButtonsToBindingsTest.kt:63)
```
I am not getting what the error is...

File: AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckDropDownAdapter.kt [hunk 1]
```diff
@@ -14,111 +14,89 @@
...
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import com.ichi2.anki.R
+import com.ichi2.libanki.Deck
+
+class DeckDropDownAdapter(private val context: Context, private val decks: List<Deck?>?) : BaseAdapter() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckDropDownAdapter.kt:28]: Could do with a `@KotlinCleanup` to make decks non-null

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
nit, but LGTM

**Review by david-allison-1 — APPROVED** [review]
Thanks!!!

**Review by mikehardy — APPROVED** [review]

**Pull request merged**